File: Js/binarySearch.js
// Binary Searching alogorithm wrtitten in JavaScript

function binarySearch(arr, key) {
    let upper = arr.length;
    let lower = 0;
    let mid = Math.floor((upper + lower)/2);
    let found = false;

    while (lower <= upper && !found) {
        if (key === arr[mid]) {
            found = true;
            return `${arr[mid]} is at position ${mid}.`
        } else if (key < arr[mid]) {
            upper = mid - 1;
        } else {
            lower = mid + 1;
        }

        mid = Math.floor((upper + lower)/2)
    }
    if (!found) return null;
}

File: Python/binary_search.py
# Binary Searching algorithm written in Python.

def binary_search(items, key):
    upper = len(items) - 1
    lower = 0
    mid = round((upper + lower)//2)
    found = False

    while lower <= upper and not found:
        if key == items[mid]:
            found = True
            return f'{items[mid]} is at position {mid}.'
        elif key < items[mid]:
            upper = mid - 1
        else:
            lower = mid + 1
        mid = round((upper + lower)//2)
    if not found: return None

